### DISCUSSION TITLE
Error when running prisma generate

### DISCUSSION THREAD

**Discussion by p1pah** [prisma/prisma#17682] [Q&A] [answered]
### Bug description

Trying to run npx prisma generate and seeing a very odd error. The error is:
```
Error: (query-engine-node-api library)
Details: Failed to convert rust `String` into napi `string`
[Context: getDmmf]
```

Some context: I am trying to help convert a very large legacy application. The database setup is very large and I am genuinely curious if this could be too big for prisma to handle. The schema.prisma is about 12000 lines long. Am I wrong in thinking that the schema might be too big for prisma generate to handle?

### How to reproduce

npx prisma generate

### Expected behavior

My prisma client would be generated successfully

### Prisma information

I'll add my schema if it warrants...

### Environment & setup

- OS: Windows
- Database:  SQL Server
- Node.js version: 16.18.1

### Prisma Version

```
4.9.0
```

**Comment by janpio** [comment, marked as answer]
In theory that is possible.
In practice we of course want to avoid that. Can you maybe share the schema with us privately? (NDA available on request of course)

**Reply by p1pah** [reply]
Where would you like it shared privately? Also, just curious, does Prisma have some tests around generating a giant schema?

**Reply by janpio** [reply]
Email to `schemas@prisma.io` would work for us.
We do have some internal test suites that work with generated (massive) schemas, but they are not our focus right now so might be overlooked.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
